snuba/datasets/errors_replacer.py

```diff
@@ -526,8 +541,42 @@ def process_message(self, message: ReplacementMessage) -> Optional[Replacement]:
 
         return processed
 
+    def _message_already_processed(self, message: ReplacementMessage) -> bool:
+        """
+        Figure out whether or not the message was already processed.
+
+        Check whether there is an entry in Redis for this specific
+        topic, consumer group, and partition. If there is, we can conclude whether
+        or not to process the incoming message by comparing the incoming message
+        to the offset in Redis.
+        """
+        key = self._build_topic_group_index_key(message.metadata)
+        processed_offset = redis_client.get(key)
```

Comment:
You do not need to read from redis at every iteration. 
This value could be useful for the current consumer only right after the start. 
Until we are caught up with messages that need to be dropped.
If an insert is successful, and the commit is successful, the following iteration does not care about what is in redis, it cannot be a message to skip. 
I think you should read it at the first iteration, keep the value in a variable, and avoid reading from redis if the variable is already populated.